lib/devices/S7.ts

```diff
@@ -63,13 +63,14 @@ export class S7Connection extends DeviceConnection {
      * Builds the S7 item group from the defined metric list
      * @param {object} vars object containing metric names and PLC addresses
      */
-    setItemGroup(vars: s7Vars) {
-        if (this.#itemGroup) {
-            this.#itemGroup.destroy();
+    addToItemGroup(vars: s7Vars) {
+        // If item group doesn't exist, create it
+        if (!this.#itemGroup) {
+            this.#itemGroup = new S7ItemGroup(this.#s7Conn);
         }
-        this.#itemGroup = new S7ItemGroup(this.#s7Conn);
         this.#itemGroup.setTranslationCB((metric: string) => this.#vars[metric]); //translates a metric name to its address
         this.#vars = vars;
```

Comment:
These two lines need changing too. The `setTranslationCB` only wants to happen once, it's part of the initialisation. Then the new `vars` need merging into `#vars`; perhaps `this.#vars = {}` in the `if` and then `this.#vars = {...this.#vars, ...vars}` here?
Properly we should be using `Map`s for dictionaries, these days, instead of objects.

Reply:
Good spot. I've pushed a tested update